### PR TITLE
JAMES-3986 AttachmentFileNameIs should be decently tested

### PR DESCRIPTION
**Pull request by chibenwa**
This changeset contributes a decent test suite with a coverage of 82% (remains logging and error handling).

It decodes file names if needed.

As such remove experimental marking.

(Needed by one of my customer)

### CODE REVIEW

**Comment by chibenwa** [comment]
@jeantil maybe?

(VN coworkers are off for the Tet)

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/AttachmentFileNameIsTest.java [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+            .matcherName("AttachmentFileNameIs")
+            .condition("file.txt")
+            .build());
+
+        assertThat(testee.match(mail))
+            .containsOnly(ANY_AT_JAMES);
+    }
+
+    @Test
+    void shouldBeCaseIncentive() throws Exception {
```
Comment (jeantil) [inline mailet/standard/src/test/java/org/apache/james/transport/matchers/AttachmentFileNameIsTest.java:356]:
```suggestion
    void shouldBeCaseInsensitive() throws Exception {
```

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/AttachmentFileNameIsTest.java [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+            .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                .setText("abc", "text/plain;\r\n name=\"fiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiile.txt\"")
+                .build())
+            .build();
+
+        AttachmentFileNameIs testee = new AttachmentFileNameIs();
+
+        testee.init(FakeMatcherConfig.builder()
+            .matcherName("AttachmentFileNameIs")
+            .condition("fiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiile.txt")
```
Comment (jeantil) [inline mailet/standard/src/test/java/org/apache/james/transport/matchers/AttachmentFileNameIsTest.java:397]: I think this test tests the core change that you did in the production code and the rest is catchup

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/AttachmentFileNameIs.java [hunk 3]
```diff
@@ -266,11 +267,10 @@ protected boolean matchFoundInZip(Part part) throws MessagingException, IOExcept
 
     /**
      * Transforms <I>fileName<I> in a trimmed lowercase string usable for matching agains the masks.
-     *
-     * @param fileName
+     * Also decode encoded words.
      */
     protected String cleanFileName(String fileName) {
-        return fileName.toLowerCase(Locale.US).trim();
+        return DecoderUtil.decodeEncodedWords(fileName.toLowerCase(Locale.US).trim(), DecodeMonitor.SILENT);
```
Comment (jeantil) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/AttachmentFileNameIs.java:273]: This seems to be the core of the change

**Review by jeantil — APPROVED** [review]

File: mailet/standard/src/test/java/org/apache/james/transport/matchers/AttachmentFileNameIsTest.java (hunk 2 above)
Comment (chibenwa) [reply mailet/standard/src/test/java/org/apache/james/transport/matchers/AttachmentFileNameIsTest.java:397]: Nope the core of the change was the encoding. Long file name was actually well handled.

Note for later: commit the test suite first, with @Disabled where necessary, and fix the code in a second commit....

**Comment by chibenwa** [comment]
Thanks Jean!

**Pull request merged**